### PR TITLE
Image seed

### LINKED ISSUE #6: Image Seeds
#### As a developer I want at least 20 images to be created when I setup the database.

Acceptance criteria:
- [x] After running `rake db:setup` at least 20 images are shown in the
  homepage.
- [x] When a new app is deployed on heroku (e.g., every review app), that app
  has at least 20 images on its homepage.

Dependencies:
- Image Index

### PR DESCRIPTION
**Pull request by BalakrishnanVasudevan**
After running rake db:setup, at least 20 images are displayed on the homepage. Script added to load review app on heroku. Fixes #6 .

### CODE REVIEW

File: db/seeds.rb [hunk 1]
```diff
@@ -1,7 +1,22 @@
-# This file should contain all the record creation needed to seed the database with its default values.
-# The data can then be loaded with the rake db:seed (or created alongside the db with db:setup).
-#
-# Examples:
-#
-#   cities = City.create([{ name: 'Chicago' }, { name: 'Copenhagen' }])
-#   Mayor.create(name: 'Emanuel', city: cities.first)
+#Image seed file
+Image.delete_all ('id <= 20')
```
Comment (bboe) [inline db/seeds.rb]: There's no need to delete these images as this only intended to run on creation.

File: db/seeds.rb [hunk 2]
```diff
@@ -1,7 +1,22 @@
...
+Image.create(url:'https://media.giphy.com/media/Uqr2RYBFdjCBq/giphy.gif', id: 1)
```
Comment (bboe) [inline db/seeds.rb]: Do you need to specify the `id` for each?

Also what happens if `create` fails? How can you ensure an exception is raised if it were to fail?

File: db/seeds.rb [hunk 3]
```diff
@@ -1,7 +1,22 @@
...
+Image.create(url:'https://media4.giphy.com/media/EU1obAC38GuWI/200.gif', id: 11)
+Image.create(url:'https://media.giphy.com/media/26tn8KQP4NewVU4Sc/giphy.gif', id: 12)
+Image.create(url:'https://media0.giphy.com/media/G50SdGptxC5uE/200w.gif', id: 13)
+Image.create(url:'https://media3.giphy.com/media/OOoiJPDijSG3u/200.gif', id: 14)
+Image.create(url:'https://media0.giphy.com/media/GDnomdqpSHlIs/200.gif', id: 15)
+Image.create(url:'https://media2.giphy.com/media/KHSIBIPQkGGWs/200.gif', id: 16)
+Image.create(url:'https://media1.giphy.com/media/F0QbYPadzSBzi/200.gif', id: 17)
+Image.create(url:'https://media.giphy.com/media/L0aWDywDu1ziw/giphy.gif', id: 18)
+Image.create(url:'https://media2.giphy.com/media/3o85g2ttYzgw6o661q/200.gif', id: 19)
+Image.create(url:'https://media.giphy.com/media/Mp4hQy51LjY6A/giphy.gif', id: 20)
```
Comment (bboe) [inline db/seeds.rb]: There's a lot of repetition here as you manually call `Image.create` 20 times. How can you simplify that process?

**Comment by bboe** [comment]
@BalakrishnanVasudevan can you set up the assignee and labels correctly for this issue/pr set?

**Comment by BalakrishnanVasudevan** [comment]
Replaced multiple image create tasks with single iterative loop to handle seeding. Removed the image delete task and removed image id. Fixes #25

File: db/seeds.rb [hunk 4]
```diff
@@ -1,7 +1,5 @@
...
+#Image seed file
+imageurls = %w('https://media.giphy.com/media/Uqr2RYBFdjCBq/giphy.gif' 'https://media.giphy.com/media/c5gYai9CzbJcc/giphy.gif' 'https://media.giphy.com/media/Zcd4GjKsg3nOw/giphy. gif' 'https://media.giphy.com/media/glmRyiSI3v5E4/giphy. gif' 'https://media.giphy.com/media/LlE3uMF20U7Ys/giphy. gif' 'https://media1.giphy.com/media/NcPQoSCgMjCTK/200. gif' 'https://media.giphy.com/media/8O9FbilJ8GxMc/giphy. gif' 'https://media3.giphy.com/media/eMxjn6fb6KyOc/200. gif' 'https://media0.giphy.com/media/12N6uT8mkxMWY0/200. gif' 'https://media1.giphy.com/media/wrzf9P70YWLJK/200. gif' 'https://media4.giphy.com/media/EU1obAC38GuWI/200. gif' 'https://media.giphy.com/media/26tn8KQP4NewVU4Sc/ giphy.gif' 'https://media0.giphy.com/media/G50SdGptxC5uE/200w. gif' 'https://media3.giphy.com/media/OOoiJPDijSG3u/200. gif' 'https://media0.giphy.com/media/GDnomdqpSHlIs/200. gif' 'https://media2.giphy.com/media/KHSIBIPQkGGWs/200. gif' 'https://media1.giphy.com/media/F0QbYPadzSBzi/200. gif' 'https://media.giphy.com/media/L0aWDywDu1ziw/giphy. gif' 'https://media2.giphy.com/media/3o85g2ttYzgw6o661q/ 200.gif' 'https://media.giphy.com/media/Mp4hQy51LjY6A/giphy.gif)
```
Comment (bboe) [inline db/seeds.rb]: The `'` characters are not needed when between `%w( ... )`. Also can you make this a little easier to read by splitting the URLs with a newline?

File: db/seeds.rb [hunk 5]
```diff
@@ -1,7 +1,5 @@
...
+imageurls.each do |link|
+  Image.create(url:link)
```
Comment (bboe) [inline db/seeds.rb]: Unless you are checking the return value, be sure to always use the ❗ form of the Rails model actions. `Image.create!`.

Also, while we're here. How many SQL queries does this result in? Is there a way to optimize this line such that it requires only a single query?

**Comment by BalakrishnanVasudevan** [comment]
Modified code to mass insert data when seeding to avoid repetitive SQL queries. Fixes #25

**Comment by BalakrishnanVasudevan** [comment]
Modified code to mass insert data when seeding to avoid repetitive SQL queries. Fixes #25 .

File: db/seeds.rb [hunk 6]
```diff
@@ -1,7 +1,26 @@
...
+               https://media3.giphy.com/media/OOoiJPDijSG3u/200.gif
+               https://media0.giphy.com/media/GDnomdqpSHlIs/200.gif
+               https://media2.giphy.com/media/KHSIBIPQkGGWs/200.gif
+               https://media1.giphy.com/media/F0QbYPadzSBzi/200.gif
+               https://media.giphy.com/media/L0aWDywDu1ziw/giphy.gif
+               https://media2.giphy.com/media/3o85g2ttYzgw6o661q/200.gif
+               https://media.giphy.com/media/Mp4hQy51LjY6A/giphy.gif }
+images = []
+imageurls.each do |u|
+  images << Image.new(url: "#{u}")
```
Comment (bboe) [inline db/seeds.rb]: `"#{u}"` is redundant

File: db/seeds.rb [hunk 7]
```diff
@@ -1,7 +1,26 @@
...
+end
+Image.import images
```
Comment (bboe) [inline db/seeds.rb]: Clever solution. You can could also do:

    Image.create!(imageurls.map { |url| { url: url } })

While the above doesn't generate a single insert statement, all of the inserts run in a single transaction.

I think your approach may be even faster. Good work!

**Comment by bboe** [comment]
Minor comment and then this is good to squash and merge.

File: db/seeds.rb (hunk 6 above)
Comment (BalakrishnanVasudevan) [inline db/seeds.rb]: Removed the redundancy.

File: db/seeds.rb (hunk 7 above)
Comment (BalakrishnanVasudevan) [inline db/seeds.rb]: Given that the number of entries being seeded is less (20), the method shows no significant improvement in terms of speed when we compare it with the older method of accessing the database multiple times. However the real advantages of this method would be visible if we had a large number of entries.

**Comment by BalakrishnanVasudevan** [comment]
Removed redundancy in the usage of a variable. After running rake db:setup, at least 20 images are displayed on the homepage. Fixes #25 .

**Comment by bboe** [comment]
👍

**Comment by bboe** [comment]
I think you forgot the squash part. Please keep that in mind for future reference.

**Pull request merged**